AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/DefaultKeyStorageInterface.dfy

```diff
@@ -0,0 +1,733 @@
+      );
+
+      var items: DDB.TransactWriteItemList := [
+        CreateTransactWritePutItem(
+          input.Version,
+          ddbTableName,
+          BRANCH_KEY_NOT_EXIST
+        ),
+        CreateTransactWritePutItem(
+          input.Active,
+          ddbTableName,
+          BRANCH_KEY_EXISTS
+        )
+      ];
+
+      var transactRequest := DDB.TransactWriteItemsInput(
+        TransactItems := items,
+        ReturnConsumedCapacity := None,
+        ReturnItemCollectionMetrics := None,
+        ClientRequestToken := None
+      );
+
+      var transactWriteItemsResponse? := ddbClient.TransactWriteItems(transactRequest);
+      var _ :- transactWriteItemsResponse?
+      .MapFailure(e => Types.ComAmazonawsDynamodb(ComAmazonawsDynamodb := e));
+
+      output := Success(Types.WriteNewEncryptedBranchKeyVersionOutput);
+
+    }
+
+    method GetEncryptedActiveBranchKey' ( input: Types.GetEncryptedActiveBranchKeyInput )
+      returns (output: Result<Types.GetEncryptedActiveBranchKeyOutput, Types.Error>)
+      requires
+        && ValidState()
+      modifies Modifies - {History}
+      // Dafny will skip type parameters when generating a default decreases clause.
+      decreases Modifies - {History}
+      ensures
+        && ValidState()
+      ensures GetEncryptedActiveBranchKeyEnsuresPublicly(input, output)
+      ensures unchanged(History)
+
+      ensures |ddbClient.History.GetItem| == |old(ddbClient.History.GetItem)| + 1
+      ensures
+        //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#getencryptedactivebranchkey
+        //= type=implication
+        //# To get the active version for the branch key id from the keystore
+        //# this operation MUST call AWS DDB `GetItem`
+        //# using the `branch-key-id` as the Partition Key and `"branch:ACTIVE"` value as the Sort Key.
+        && Seq.Last(ddbClient.History.GetItem).input.Key
+           == map[
+                Structure.BRANCH_KEY_IDENTIFIER_FIELD := DDB.AttributeValue.S(input.Identifier),
+                Structure.TYPE_FIELD := DDB.AttributeValue.S(Structure.BRANCH_KEY_ACTIVE_TYPE)
+              ]
+
+      ensures output.Success?
+              ==>
+                && Seq.Last(ddbClient.History.GetItem).output.Success?
+                && Seq.Last(ddbClient.History.GetItem).output.value.Item.Some?
+                   //= aws-encryption-sdk-specification/framework/key-store/dynamodb-key-storage.md#getencryptedactivebranchkey
+                   //= type=implication
+                   //# The AWS DDB response MUST contain the fields defined in the [branch keystore record format](#record-format).
+                && Structure.BranchKeyItem?(Seq.Last(ddbClient.History.GetItem).output.value.Item.value)
+
+      ensures
+        && old(ddbClient.History.GetItem) < ddbClient.History.GetItem
+        && old(ddbClient.History.TransactWriteItems) == ddbClient.History.TransactWriteItems
```

Comment:
@seebees What is going on here?

I just stumbled on this while working on Mutations,
but why assert the history of TransactWriteItems? 

Is this a copy-paste mistake?